File: api/src/endpoint/repository/get_repository_tree.rs
use super::default_ref;
use crate::{endpoint::Endpoint, resource::repository::RepositoryTreeResource};
use serde::{Deserialize, Serialize};

pub struct GetRepositoryTree;

impl Endpoint for GetRepositoryTree {
    const PATH: &'static str = "/repository/{owner}/{repo}/tree";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryTreeRequest;
    type Response = GetRepositoryTreeResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetRepositoryTreeRequest {
    #[serde(default = "default_ref")]
    pub ref_name: String,
}

pub type GetRepositoryTreeResponse = RepositoryTreeResource;

File: gitdot-api-ts/src/endpoint/repository/get-repository-tree.ts
import { z } from "zod";

import { RepositoryTreeResource } from "../../resource";

export const GetRepositoryTreeRequest = z.object({
  ref_name: z.string(),
});
export type GetRepositoryTreeRequest = z.infer<typeof GetRepositoryTreeRequest>;

export const GetRepositoryTree = {
  path: "/repository/{owner}/{repo}/tree",
  method: "GET",
  request: GetRepositoryTreeRequest,
  response: RepositoryTreeResource,
} as const;
export type GetRepositoryTree = typeof GetRepositoryTree;
